packages/bbui/src/Actions/click_outside.ts

```diff
@@ -91,9 +91,19 @@ const handleMouseDown = (e: MouseEvent) => {
   document.addEventListener("click", handleMouseUp, true)
 }
 
+// Handle iframe clicks by detecting a loss of focus on the main window
+const handleBlur = () => {
+  if (document.activeElement?.tagName === "IFRAME") {
+    handleClick(
+      new MouseEvent("click", { relatedTarget: document.activeElement })
+    )
+  }
+}
+
 // Global singleton listeners for our events
 document.addEventListener("mousedown", handleMouseDown)
 document.addEventListener("contextmenu", handleClick)
+window.addEventListener("blur", handleBlur)
```

Comment:
Should this be in bbui, or should it be part of the builder only?

Reply:
I don't think it matters if it's in BBUI - all it'll do is trigger an additional click (with an iframe being the target) if focus shifts to an iframe element. I think it's valid to include this everywhere, as otherwise iframe clicks are always lost (e.g. if inside a client app and clicking on an iframe - today we would not catch this, but we would now).